app/src/main/java/com/kamron/pogoiv/MainActivity.java

```diff
@@ -295,13 +295,42 @@ private void startPokeFly() {
         Intent intent = Pokefly.createIntent(this, trainerLevel, statusBarHeight, batterySaver);
         startService(intent);
 
-        pokeFlyRunning = true;
-
         if (settings.shouldLaunchPokemonGo()) {
             openPokemonGoApp();
         }
     }
 
+    private boolean isPokeFlyRunning() {
+        ActivityManager manager = (ActivityManager) getSystemService(Context.ACTIVITY_SERVICE);
+        for (ActivityManager.RunningServiceInfo serviceInfo : manager.getRunningServices(Integer.MAX_VALUE)) {
+            if (Pokefly.class.getName().equals(serviceInfo.service.getClassName())) {
+                return true;
+            }
+        }
```

Comment:
FWIW we used to use some flag, not sure it worked well.

> boolean static variabled that you set/unset when the service starts/stops.

Not static please. If new instances of the Activity are created they shouldn't keep any old value. Docs aren't clear that happens, but posts like http://android-developers.blogspot.de/2009/01/avoiding-memory-leaks.html seem to imply that's possible, and so does e.g. https://developer.android.com/training/basics/activity-lifecycle/recreating.html. Not that the docs are very explicit.
Getting a good answer on StackOverflow seems impossible though.